File: LayoutTests/http/tests/loading/authentication-after-redirect-stores-wrong-credentials/resources/wrong-credential-3-output-credentials-then-finish.php
<?php
if (!isset($_SERVER['PHP_AUTH_USER']))
	echo "No HTTP authentication credentials<br>";
else
	echo "Authenticated as {$_SERVER['PHP_AUTH_USER']} with password {$_SERVER['PHP_AUTH_PW']}<br>";
?>
<script>
if (window.layoutTestController)
	layoutTestController.notifyDone();
</script>

File: LayoutTests/http/tests/loading/authentication-after-redirect-stores-wrong-credentials/resources/wrong-credential-3-output-credentials-then-finish.py
#!/usr/bin/env python3

import base64
import os
import sys

credentials = base64.b64decode(os.environ.get('HTTP_AUTHORIZATION', ' Og==').split(' ')[1]).decode().split(':')
username = credentials[0]
password = ':'.join(credentials[1:])

sys.stdout.write('Content-Type: text/html\r\n\r\n')

if not username:
    sys.stdout.write('No HTTP authentication credentials<br>')
else:
    sys.stdout.write('Authenticated as {} with password {}<br>'.format(username, password))

sys.stdout.write(
    '<script>\n'
    'if (window.testRunner)\n'
    '    testRunner.notifyDone();\n'
    '</script>\n'
)